### DISCUSSION TITLE
Have nvim-java configure jdtls to use the OpenJDK17 dependency explicitly

### DISCUSSION THREAD

**Discussion by neandrake** [nvim-java/nvim-java#209] [Ideas]
I'm getting nvim-java setup on both Windows and Mac systems and seem to have gotten things into a good working state except on one mac. When jdtls starts up on that system it errors out indicating that java 17 is required, even though nvim-java is installing OpenJDK 17. It turns out on that mac system I have `JAVA_HOME` set to a Java 11 installation which is required for the projects I'm working on.

Is it possible for nvim-java to configure/invoke jdtls pointing it to the OpenJDK 17 installation installed as dependency, rather than jdtls resolving which installation to use on its own? I'm going to pick at my config to see if I can configure jdtls to point to the 17 install but it should be reasonable for nvim-java to explicitly configure jdtls to use this dependency. If I do get things working and I figure out nvim-java internals I can try to create a PR!

**Comment by s1n7ax** [comment]
@neandrake Let's say we have java 11 in the PATH env variable already. When you launch neovim and run `:!java --version`, it always first check the mason bin first. So if jdk17 is installed, this will be picked up first.

But in your cast, it's obviously not happening.

Some time ago, we fixed mason package path issue where I'm linking a wrong path in macOs. You can give it a try now after updating all the plugins. and then try removing  openjdk-17 from mason to force it to reinstall. This should work. If not let me know
